src/helpers/figures/charts/scorecard_chart_config_builder.ts

```diff
@@ -0,0 +1,379 @@
+import { Color } from "chart.js";
+import { DEFAULT_FONT } from "../../../constants";
+import { Figure, Pixel, PixelPosition, Style } from "../../../types";
+import { ScorecardChartRuntime } from "../../../types/chart";
+import { relativeLuminance } from "../../color";
+import { getFontSizeMatchingWidth } from "../../text_helper";
+
+/* Sizes of boxes containing the texts, in percentage of the Chart size */
+const TITLE_FONT_SIZE = 18;
+
+const BASELINE_BOX_HEIGHT_RATIO = 0.35;
+const KEY_BOX_HEIGHT_RATIO = 0.65;
+
+/** Baseline description should have a smaller font than the baseline */
+const BASELINE_DESCR_FONT_RATIO = 0.9;
+
+/* Padding at the border of the chart, in percentage of the chart width */
+const CHART_PADDING_RATIO = 0.02;
+
+/**
+ * Line height (in em)
+ * Having a line heigh =1em (=font size) don't work, the font will overflow.
+ */
+const LINE_HEIGHT = 1.2;
+
+type ScorecardChartElement = {
+  text: string;
+  style: {
+    font: string;
+    color: Color;
+    strikethrough?: boolean;
+    underline?: boolean;
+  };
+  position: PixelPosition;
+};
+
+export type ScorecardChartConfig = {
+  canvas: {
+    width: number;
+    height: number;
+    backgroundColor: Color;
+  };
+  title?: ScorecardChartElement;
+  baselineArrow?: ScorecardChartElement;
+  baseline?: ScorecardChartElement;
+  baselineDescr?: ScorecardChartElement;
+  key?: ScorecardChartElement;
+};
+
+export function formatBaselineDescr(
+  baselineDescr: string | undefined,
+  baseline: string | undefined
+): string {
+  const _baselineDescr = baselineDescr || "";
+  return baseline && _baselineDescr ? " " + _baselineDescr : _baselineDescr;
+}
+
+function getDefaultContextFont(
+  fontSize: number,
+  bold: boolean | undefined = false,
+  italic: boolean | undefined = false
+): string {
+  const italicStr = italic ? "italic" : "";
+  const weight = bold ? "bold" : "";
+  return `${italicStr} ${weight} ${fontSize}px ${DEFAULT_FONT}`;
+}
+
+export function getScorecardConfiguration(
+  figure: Figure,
```

Comment:
I still have a problem with this. My take was that I really don't think that we should have to pass figure as a props to the chart component because the chart should be adapted to its container size without having to know what it's container is.
I'd rather have getScorecardConfiguration take {width, height} as argument.